libs-haskell/bazel-runfiles/src/DA/Bazel/Runfiles.hs

```diff
@@ -87,8 +98,8 @@ locateResource Resource { runfilesPath, resourcesPath } = do
 locateRunfiles :: HasCallStack => FilePath -> IO FilePath
 locateRunfiles runfilesPath =
   locateResource Resource
-    { runfilesPath
-    , resourcesPath = ""
+    { resourcesPath = ""
+    , runfilesPathPrefix = runfilesPath
```

Comment:
I imagine you need `dirname` here

Reply:
No, to see the problem consider

```haskell
locateRunfiles "WORKSPACE/TARGET/PATH/TO/FILE.EXT"
```

on a Bazel target, this would return `"PATH_TO_RUNFILES/WORKSPACE/TARGET/PATH/TO/FILE.EXT"`, while on a packaged app it would just return `"PATH_TO_EXECUTABLE/../resources/"`.

With the definition of `locateRunfiles` here, this would be equivalent to the following `locateResource` call:

```haskell
locateResource Resource { runfilesPathPrefix = "WORKSPACE/TARGET/PATH/TO/FILE.EXT", resourcesPath = "" } 
```

Meanwhile, the version with `takeDirectory` (equivalent to bash `dirname`) would result in

```haskell
locateResource Resource { runfilesPathPrefix = "WORKSPACE/TARGET/PATH/TO/", resourcesPath = "" }
```

which on a bazel target would return `"PATH_TO_RUNFILES/WORKSPACE/TARGET/PATH/TO"`, missing `/FILE.EXT` compared with the expected output.

Using `splitDirectory`, which returns a tuple with the equivalent of `dirname` and `basename`, this would result in

```haskell
locateResource Resource { runfilesPathPrefix = "WORKSPACE/TARGET/PATH/TO/", resourcesPath = "FILE.EXT" }
```

which would have the right behavior on a bazel target, but would return `"PATH_TO_EXECUTABLE/../resources/FILE.EXT"`, adding `/FILE.EXT` compared with the expected output. 

This last point isn't as important because another change in this PR is to replace all uses of `locateRunfiles` with `locateResource` when they might be used from a packaged app, but in that case there's no benefit over the version with `{ resourcesPath = "" }`, and the path manipulation might be misleading.

Reply:
Thanks for explaining!